### PR TITLE
Added Rat Tumour Example

### PR DESCRIPTION
**Pull request by Dpananos**
Added notebook for rat tumor example in chapter 5 of BDA.

I was looking for some resources in python for Gelman's first example of hierarchical models in BDA but couldn't find any.  I decided to write one myself.

I'd appreciate any feedback re: this example.

### CODE REVIEW

**Comment by junpenglao** [comment]
Looks nice! A few comments:
- Please index the notebook under https://github.com/pymc-devs/pymc3/blob/master/docs/source/examples.rst#glm
- There are some guideline about unifying the style of the docs, please have a look at this issue: https://github.com/pymc-devs/pymc3/issues/2834
- you can put `;` at the end of plotting command suppresses the text output: `pm.traceplot(trace);`
- you should use the default initialization, also shorter trace is sufficient:
`trace = pm.sample(50_000, step = pm.NUTS())` --> `trace = pm.sample(1000, tune=2000)`
- Transformed is needed for the Densitydist as the parameter is defined on (0, inf), which is why on the discourse answer I supplied `transform=tr.log` https://discourse.pymc.io/t/reparametizing-gelmans-rat-tumour-example/1417/3?u=junpenglao. Otherwise you will get a divergence when a proposal went out of bound.
- use `import theano.tensor as tt`, `tt` is mostly used in our docs
- please write `p = pm.Binomial('y', n=n_rat, p=theta, observed=n_tumour)` which is more conventional.

**Comment by fonnesbeck** [comment]
Also, 50,000 samples from NUTS is far too many. Around 1000 is usually sufficient, unless there are effective sample size issues. You can add iterations to the `tune` argument if you are worried about convergence, but once converged, you don't need many samples.

**Comment by Dpananos** [comment]
I've omitted a discussion about transforms (as per your request on discourse) because the example no longer uses `DensityDist`.  If you still feel an explanation is required, I'd be happy to include it.  I just need to know what you'd like me to write.

Alternatively, we could use `DensityDist` in the example, but I would need a solution such that the estimates are not biased.

File: docs/source/examples.rst [hunk 1]
```diff
@@ -35,6 +35,7 @@ GLM
 ===
 
 .. toctree::
+   notebooks/RatTumorExample.ipynb
```
Comment (junpenglao) [inline docs/source/examples.rst]: Could you please move this to the last session of GLM, also change the title of the notebook:
`rat tumor`--> `Hierarchical Binominal Model: Rat Tumor Example`

File: docs/source/notebooks/RatTumorExample.ipynb [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+    "\n",
+    "$$ y_i \\sim \\operatorname{Bin}(\\theta_i;n_i)$$\n",
+    "\n",
+    "allowing the probability of developing an endometrial stromal polyp (i.e. $\\theta_i$) to be drawn from some population distribution.  For analytical tractability, we assume that $\\theta_i$ has Beta distribution\n",
+    "\n",
+    "$$ \\theta_i \\sim \\operatorname{Beta}(\\alpha, \\beta)\\>. $$\n",
+    "\n",
+    "We are free to specify a prior distribution for $\\alpha, \\beta$.  We choose a weakly informative prior distribution to reflect our ignorance about the true values of $\\alpha, \\beta$.  The authors of BDA3 choose the joint hyperprior for $\\alpha, \\beta$ to be\n",
+    "\n",
+    "$$ p(\\alpha, \\beta) \\propto (\\alpha + \\beta) ^{-5/2} \\>.$$\n",
```
Comment (junpenglao) [inline docs/source/notebooks/RatTumorExample.ipynb]: The math rendering looks a bit odd, i believe `\\>.` are not needed:
screen shot 2018-06-27 at 8 06 38 am

**Comment by Dpananos** [comment]
Changes made.

File: docs/source/notebooks/GLM-hierarchical-binominal-model.ipynb [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# A Directly Computed Solution\n",
+    "\n",
+    "Our joint posterior distribution is\n",
+    "\n",
+    "$$p(\\alpha,\\beta,\\theta \\lvert y) \n",
+    "\\propto \\>\n",
```
Comment (junpenglao) [inline docs/source/notebooks/GLM-hierarchical-binominal-model.ipynb]: a few `\\>` still need to be removed.

Comment (Dpananos) [reply docs/source/notebooks/GLM-hierarchical-binominal-model.ipynb]: Got em

**Comment by junpenglao** [comment]
Awesome, thanks!

**Pull request merged**